### ISSUE TITLE
[transformer] Choice of Reference Optimizer

### ISSUE THREAD

**Issue by nvcforster** [mlcommons/training_policies#49]
The translation reference implementation currently uses the Lazy Adam sparse optimizer, but this becomes less effective when going to multi-device and/or multi-node as batch sizes become larger and sparsity decreases. The decreased sparsity and cost of storing/retrieving indices to access the data causes the overhead of sparsity to outweigh the benefits. Additionally, it seems preferable for frameworks to do an all-reduce on gradients in parallel training rather than an all-gather with sparse gradients to handle repeating indices. The request is to move to a non-sparse Adam optimizer in the reference implementation.

**Comment by bitfort** [comment]
SWG: 

To clarify -- this refers to Transformer. 

We will revisit next week with results for convergence with the new Lazy Adam Dense Optimizer. Further discussion.

**Comment by bitfort** [comment]
SWG:

Closing as a duplicate of the current discussion on the general Adam versus LazyAdam.
